src/main/java/hudson/plugins/parameterizedtrigger/CollectionUtils.java

```diff
@@ -0,0 +1,29 @@
+package hudson.plugins.parameterizedtrigger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Internal helpers that should be replaced by Java11 equivalents when the JDK baseline moves on.
+ *
+ */
+@Restricted(NoExternalUse.class)
+final class CollectionUtils {
+
+    static <T> List<T> immutableList(Collection<T> collection1, Collection<T> collection2) {
+        List<T> list = new ArrayList<>(collection1);
+        list.addAll(collection2);
```

Comment:
Since we know the sizes of both lists it is potentially more effective to initialize the list with the correct capacity to avoid unnecessary extra array copying.

 ```suggestion
        List<T> list = new ArrayList<>(collection1.size() + collection2.size());
        list.addAll(collection1);
        list.addAll(collection2);
```

Reply:
addressed in 62f34046e272974f0a78de1242b58b1830c051e8